metaci/release/tests/test_tasks.py

```diff
@@ -420,3 +422,250 @@ def test_release_merge_freeze_if_safe__safe(smfs_mock):
+    )
+
+    execute_active_release_cohorts()
+    rc.refresh_from_db()
+    rc_progress.refresh_from_db()
+
+    assert rc.status == ReleaseCohort.STATUS.completed
+    assert rc_progress.status == ReleaseCohort.STATUS.active
+
+
+@pytest.mark.django_db
+@unittest.mock.patch("metaci.release.tasks.advance_releases")
+@unittest.mock.patch("metaci.release.tasks.set_merge_freeze_status")
+def test_execute_active_release_cohorts__advances_release_cohorts(
+    smfs_mock,
+    advance_releases_mock,
+):
+    rc = ReleaseCohortFactory(status=ReleaseCohort.STATUS.active, dependency_graph={})
+    _ = ReleaseFactory(
+        repo__url="foo", release_cohort=rc, status=Release.STATUS.inprogress
+    )
+    _ = ReleaseCohortFactory(status=ReleaseCohort.STATUS.completed, dependency_graph={})
+
+    execute_active_release_cohorts()
+
+    advance_releases_mock.assert_called_once_with(rc)
+
+
+@pytest.mark.django_db
+@unittest.mock.patch("metaci.release.tasks.set_merge_freeze_status")
+def test_get_dependency_graph(smfs_mock):
+    # Create a simulated repo graph that includes:
+    # Two repos depending on the same base, one of which also requires the other.
+    # A repo with no other relationships
+
+    # Create database entries
+    rc = ReleaseCohortFactory(status=ReleaseCohort.STATUS.active, dependency_graph={})
+    top = ReleaseFactory(
+        repo__url="https://github.com/example/top",
+        release_cohort=rc,
+        status=Release.STATUS.draft,
+        created_from_commit="abc",
+    )
+    left = ReleaseFactory(
+        repo__url="https://github.com/example/left",
+        release_cohort=rc,
+        status=Release.STATUS.draft,
+        created_from_commit="ghi",
+    )
+    right = ReleaseFactory(
+        repo__url="https://github.com/example/right",
+        release_cohort=rc,
+        status=Release.STATUS.draft,
+        created_from_commit="jkl",
+    )
+    separate = ReleaseFactory(
+        repo__url="https://github.com/example/separate",
+        release_cohort=rc,
+        status=Release.STATUS.draft,
+        created_from_commit="mno",
+    )
+
+    # Mock results from GitHubDynamicDependency.flatten()
+    flatten_results = {
+        top.repo.url: [
+            UnmanagedGitHubRefDependency(
+                github=top.repo.url,
+                ref=top.created_from_commit,
+                subfolder="unpackaged/pre/first",
+            ),
+        ],
+        left.repo.url: [GitHubDynamicDependency(github=top.repo.url)],
+        right.repo.url: [
+            GitHubDynamicDependency(github=top.repo.url),
+            GitHubDynamicDependency(github=left.repo.url),
+        ],
+        separate.repo.url: [],
+    }
+
+    def flatten_mock(self, context):
+        return flatten_results[self.github]
+
+    with unittest.mock.patch.object(GitHubDynamicDependency, "flatten", flatten_mock):
+        result = get_dependency_graph([top, left, right, separate])
```

Comment:
Let's add a second call to `get_dependency_graph` where the cohort does not include `top` and assert that the returned graph is still the same.